_docker/drupal/drupal-filesystem/web/themes/custom/rhdp2/rhdp2.theme

```diff
@@ -86,113 +87,196 @@ function rhdp2_preprocess_node(&$variables) {
-
-        $variables['authors'] = $authors_link;
+        $authors_values = [];
+
+        foreach ($author_nodes as $author_node) {
+          $authors_values[] = $author_node->getTitle();
+        }
+
+        // Keep spaces inserted around & for output.
+        $variables['authors'] = implode(' & ', $authors_values);
       }
     }
-    else {
-      $authors_values = [];
 
-      foreach ($author_nodes as $author_node) {
-        $authors_values[] = $author_node->getTitle();
-      }
+    // Process for view mode compact_dynamic_article_list_item.
+    if ($variables['view_mode'] === 'compact_dynamic_article_list_item') {
+      $variables['rhd_domain'] = FALSE;
+      $variables['absolute_url'] = FALSE;
 
-      $variables['authors'] = implode('&', $authors_values);
-    }
-  }
-  if ($type === 'article' && $variables['view_mode'] === 'compact_dynamic_article_list_item') {
-    $variables['rhd_domain'] = FALSE;
-    $variables['absolute_url'] = FALSE;
-
-    $current_path = \Drupal::service('path.current')->getPath();
-    $url = Url::fromUri("internal:$current_path");
-    $variables['absolute_url'] = $url->setAbsolute()->toString();
-
-    // don't run comment js on this entity
-    // if it isn't a redhat domain or the current domain
-    if ($_SERVER['SERVER_NAME']) {
-      $host = parse_url($variables['absolute_url'], PHP_URL_HOST);
-      if ($host === $_SERVER['SERVER_NAME'] && !(strpos($host, 'redhat.com') === FALSE)) {
-        $variables['rhd_domain'] = TRUE;
+      $current_path = \Drupal::service('path.current')->getPath();
+      $url = Url::fromUri("internal:$current_path");
+      $variables['absolute_url'] = $url->setAbsolute()->toString();
+
+      // Requires redhat domain to run comment js.
+      if ($_SERVER['SERVER_NAME']) {
+        $host = parse_url($variables['absolute_url'], PHP_URL_HOST);
+        if ($host === $_SERVER['SERVER_NAME'] && !(strpos($host, 'redhat.com') === FALSE)) {
+          $variables['rhd_domain'] = TRUE;
+        }
       }
     }
   }
 
   // Author content type.
-  if ($type == 'author') {
+  if ($type === 'author') {
     $variables['author_name'] = $node->getTitle();
     $variables['author_articles'] = FALSE;
+    $variables['social_links'] = FALSE;
+
     if (views_get_view_result('author_articles')) {
       $variables['author_articles'] = views_embed_view('author_articles');
     }
-    $variables['social_links'] = FALSE;
+
     $social_facebook = $node->get('field_facebook')->getValue();
     $social_github = $node->get('field_github')->getValue();
     $social_linkedin = $node->get('field_linkedin')->getValue();
     $social_twitter = $node->get('field_twitter')->getValue();
     $social_youtube = $node->get('field_youtube')->getValue();
-    $variables['social_github'] = ($social_github) ? $social_github[0]['uri'] : FALSE;
+
+    // @TOD: Is this meant to be outside of the social_links true logic?
+    $variables['social_github'] = $social_github[0]['uri'] ?? FALSE;
```

Comment:
We should move this into the conditional. I am not sure why it is separated from the other social links